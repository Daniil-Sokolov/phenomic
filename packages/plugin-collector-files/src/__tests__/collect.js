import db from "@phenomic/core/lib/db";

import plugin from "..";

import fixtures from "./__fixtures__";

it("should collect stuff", async () => {
  db.destroy();

  const p = plugin();
  Object.keys(fixtures).map(path => {
    p.collect(db, path, fixtures[path]);
  });

  expect(await db.getList("__null__")).toMatchSnapshot();
  expect(await db.getList("news/2017")).toMatchSnapshot();
  expect(await db.getList("showcaseTags")).toMatchSnapshot();
});
