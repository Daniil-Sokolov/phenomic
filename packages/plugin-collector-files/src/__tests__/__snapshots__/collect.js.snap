// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should collect stuff 1`] = `
Array [
  Object {
    "filename": "open-source.md",
    "id": "showcaseTags/open-source",
    "name": "Open Source",
  },
  Object {
    "filename": "acrowithbrian.com.md",
    "id": "showcase/entry/acrowithbrian.com",
    "showcaseTags": Array [
      "event",
      "blog",
      "learning",
      "business",
    ],
    "source": "https://github.com/brianrc/acrowithbrian-site",
    "title": "Brian Cruikshank Acro",
    "url": "https://www.acrowithbrian.com/",
  },
  Object {
    "filename": "anthonydugois.com__polynomic.md",
    "id": "showcase/entry/anthonydugois.com__polynomic",
    "showcaseTags": Array [
      "docs",
      "open-source",
    ],
    "source": "https://github.com/anthonydugois/polynomic",
    "title": "Polynomic",
    "url": "http://anthonydugois.com/polynomic/",
  },
  Object {
    "filename": "ced.photo.md",
    "id": "showcase/entry/ced.photo",
    "showcaseTags": Array [
      "blog",
      "business",
    ],
    "source": "https://github.com/xuopled/ced.photo",
    "title": "CeD photo",
    "url": "https://www.ced.photo",
  },
  Object {
    "filename": "submit.md",
    "id": "showcase/submit",
    "title": "Submit a website/app made with Phenomic to the Showcase",
  },
  Object {
    "bio": "moox short bio",
    "body": "moox long bio",
    "filename": "moox.json",
    "id": "authors/moox",
    "name": "moox",
  },
  Object {
    "bio": "bloodyowl short bio",
    "body": "bloodyowl long bio",
    "filename": "bloodyowl.json",
    "id": "authors/bloodyowl",
    "name": "bloodyowl",
  },
  Object {
    "filename": "event.md",
    "id": "showcaseTags/event",
    "name": "Event",
  },
  Object {
    "authors": Array [
      "bloodyowl",
    ],
    "date": "2017-06-02T00:00:00.000Z",
    "filename": "introducing-1.0.0-alpha.md",
    "id": "news/2017/06/introducing-1.0.0-alpha",
    "title": "Introducing Phenomic 1.0.0 first alpha",
  },
]
`;

exports[`should collect stuff 2`] = `
Array [
  Object {
    "authors": Array [
      "bloodyowl",
    ],
    "date": "2017-06-02T00:00:00.000Z",
    "filename": "introducing-1.0.0-alpha.md",
    "id": "06/introducing-1.0.0-alpha",
    "title": "Introducing Phenomic 1.0.0 first alpha",
  },
]
`;

exports[`should collect stuff 3`] = `
Array [
  Object {
    "id": "blog",
  },
  Object {
    "id": "business",
  },
  Object {
    "id": "docs",
  },
  Object {
    "id": "learning",
  },
  Object {
    "filename": "open-source.md",
    "id": "open-source",
    "name": "Open Source",
  },
  Object {
    "filename": "event.md",
    "id": "event",
    "name": "Event",
  },
]
`;
